scenario.py

```diff
@@ -189,10 +198,204 @@ def __init__(self, params, experiment_path):
 
         self.save_folder.mkdir(parents=True, exist_ok=True)
 
+        # ------------------------------------------------
+        # Print the description of the scenario configured
+        # ------------------------------------------------
+
+        # Describe scenario
+        print("\n### Description of data scenario configured:")
+        print("- Number of partners defined:", self.partners_count)
+        print("- Data distribution scenario chosen:")
+        print("   - Simple split option:", self.samples_split_option)
+        print("   - Advanced split:", self.advanced_split)
+        print("- Test data distribution scenario chosen:", self.single_partner_test_mode)
+        print("- Weighting option:", self.aggregation_weighting)
+        print(
+            "- Number of epochs and mini-batches: "
+            + str(self.epoch_count)
+            + " epochs and "
+            + str(self.minibatch_count)
+            + " mini-batches"
+        )
+
+        # Describe data
+        print("\n### Data loaded: ", self.dataset_name)
+        print("- " + str(len(self.x_train)) + " train data with " + str(len(self.y_train)) + " labels")
+        print("- " + str(len(self.x_val)) + " val data with " + str(len(self.y_val)) + " labels")
+        print("- " + str(len(self.x_test)) + " test data " + str(len(self.y_test)) + " labels")
+
     def append_contributivity(self, contributivity):
 
         self.contributivity_list.append(contributivity)
 
+    def split_data_advanced(self):
+        """Advanced split: Populates the partners with their train and test data (not pre-processed)"""
+
+        x_train = self.x_train
+        y_train = self.y_train
+        x_test = self.x_test
+        y_test = self.y_test
+        amounts_per_partner = self.amounts_per_partner
+        advanced_split = self.advanced_split
+
+        # Compose the lists of partners with data samples from shared clusters and those with specific clusters
+        shared = [(idx, x[0]) for idx, x in enumerate(advanced_split) if x[1] == 'shared']
+        specific = [(idx, x[0]) for idx, x in enumerate(advanced_split) if x[1] == 'specific']
+
+        # Compose the list of different labels/classes in the dataset
+        classes = list(set(y_train))
+        random.seed(42)
+        random.shuffle(classes)
+
+        # Check coherence of the split option:
+        nb_diff_classes = len(classes)
+        assert sum([x[1] for x in specific]) + max([x[1] for x in shared]) <= nb_diff_classes
+
+        # Stratify the dataset per labels/classes
+        x_train_for_cluster, y_train_for_cluster, count_available_per_cluster = {}, {}, {}
+        for label in classes:
```

Comment:
use only label or only class. I guess label would be easier since class is a python keyword.

Reply:
Done